transports/noise/tests/smoke.rs

```diff
@@ -50,8 +50,8 @@ fn xx() {
 
         futures::executor::block_on(async move {
             let (
-                (reported_client_id, mut client_session),
-                (reported_server_id, mut server_session),
+                (reported_client_id, mut server_session),
+                (reported_server_id, mut client_session),
```

Comment:
Is this diff necessary? Is it even correct?

Reply:
Yes it is correct, it was wrong before.

Reply:
The change makes sense to me. For the record, this test did not fail before as the roles were simply switched, which does not impact the test execution below.


Nice catch @oblique. In general, preference for these unrelated changes to be in separate pull requests. Given that this is rather small, fine to merge here.